src/main/java/com/google/sps/model/AuthenticatedHttpServlet.java

```diff
@@ -37,18 +56,24 @@
   @Override
   public final void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException {
-    Credential googleCredential = AuthenticationUtility.getGoogleCredential(request);
-    if (googleCredential == null) {
+    try {
+      Credential googleCredential = getGoogleCredential(request);
+      if (googleCredential == null) {
+        response.sendError(403, ERROR_403);
+        return;
+      }
+      doGet(request, response, googleCredential);
+    } catch (TokenVerificationError e) {
       response.sendError(403, ERROR_403);
-      return;
+    } catch (GeneralSecurityException | IOException e) {
+      response.sendError(500, e.getMessage());
```

Comment:
The default behavior for uncaught exceptions should be to return an error 500. What's your motivation for explicitly catching these?

Separately, it's not a good idea to return the message of exceptions that you didn't create yourself. You don't know what kind of internal implementation details are in that message.

Reply:
Because I'm overriding doGet and doPost, they are originally declared with throwing IOException. Java won't let me add to this list of errors thrown in an override, and I cannot change the original definition. I also have to either catch or throw this error (and add it to the method signature) or I'll get a compiler error. 

An IOException (or any unchecked exception) will default to a 500, but I have to catch the GeneralSecurityException. I can. however, throw a ServletException, which is designed for use when the request cannot be handled. 

I just changed the error message, removed the catch expression for the IOException (as this can be thrown from a doGet or doPost), and threw the new error type.